scene/animation/animation_player.cpp

```diff
@@ -33,6 +33,17 @@
 #include "message_queue.h"
 #include "scene/scene_string_names.h"
 
+#ifdef TOOLS_ENABLED
+void AnimatedValuesBackup::update_skeletons() {
+
+	for (int i = 0; i < entries.size(); i++) {
+		if (entries[i].bone_idx != -1) {
+			Object::cast_to<Skeleton>(entries[i].object)->notification(Skeleton::NOTIFICATION_UPDATE_SKELETON);
+		}
+	}
+}
+#endif
```

Comment:
technically, you should be able to call MessageQueue::flush() for this to happen, though make sure that you are not called from this function when doing it.

Reply:
I did it that way because I wasn't sure no unrelated messages that can interfere somehow with the process are in the queue by that point.

Furthermore, this is called during a deferred call to `_prepare_onion_layers_2()` so we are already there.